### PR TITLE
Fixes the friction computation

### PR DESCRIPTION
**Pull request by bishtgautam**
Corrects the v-velocity and use of Manning's coefficient. Also, saves maximum Courant number
instead of max wave speed.

### CODE REVIEW

File: src/physics_swe.c [hunk 1]
```diff
@@ -267,15 +267,16 @@ static PetscErrorCode RHSFunctionForInternalEdges(RDy rdy, Vec F, AMaxDiagnostic
...
-      if (amax_vec_int[ii] > amax_diags->max_wave_speed) {
-        amax_diags->max_wave_speed = amax_vec_int[ii];
-        amax_diags->global_edge_id = edges->global_ids[ii];
-      }
-
       if (!(hr < tiny_h && hl < tiny_h)) {
         PetscReal areal = cells->areas[l];
         PetscReal arear = cells->areas[r];
 
+        PetscReal cnum = amax_vec_int[ii] * edge_len / fmin(areal, arear) * rdy->dt;
```
Comment (bishtgautam) [inline src/physics_swe.c:274]: This is based on https://github.com/RDycore/toy-problems/blob/main/swe_roe/ex2b.c#L1670-1671

File: src/physics_swe.c [hunk 2]
```diff
@@ -374,10 +375,12 @@ static PetscErrorCode RHSFunctionForBoundaryEdges(RDy rdy, Vec F, AMaxDiagnostic
         PetscReal hl = x_ptr[icell * ndof + 0];
 
         if (!(hl < tiny_h)) {
-          if (amax_vec_bnd[e] > amax_diags->max_wave_speed) {
-            amax_diags->max_wave_speed = amax_vec_bnd[e];
-            amax_diags->global_edge_id = edges->global_ids[e];
+          PetscReal cnum = amax_vec_bnd[icell] * edge_len / cell_area * rdy->dt;
+          if (cnum > courant_num_diags->max_courant_num) {
+            courant_num_diags->max_courant_num = cnum;
```
Comment (bishtgautam) [inline src/physics_swe.c:380]: This is based on https://github.com/RDycore/toy-problems/blob/main/swe_roe/ex2b.c#L1792

File: src/physics_swe.c [hunk 3]
```diff
@@ -13,56 +13,56 @@ static const PetscReal GRAVITY = 9.806;
 // which the maximum wave speed is encountered. If you change this struct,
 // update the call to MPI_Type_create_struct in InitMPITypesAndOps below.
 typedef struct {
-  PetscReal max_wave_speed;  // maximum wave speed encountered
-  PetscInt  global_edge_id;  // edge at which the max wave speed was encountered
-} AMaxDiagnostics;
+  PetscReal max_courant_num;  // maximum wave speed encountered
```
Comment (jeff-cohere) [inline src/physics_swe.c:16]: You probably want to update the comment here

File: src/physics_swe.c [hunk 4]
```diff
@@ -435,7 +438,7 @@ static PetscErrorCode AddSourceTerm(RDy rdy, Vec F) {
       PetscReal bedy = dz_dy * GRAVITY * h;
 
       PetscReal u = u_vec[icell];
-      PetscReal v = u_vec[icell];
+      PetscReal v = v_vec[icell];
```
Comment (jeff-cohere) [inline src/physics_swe.c:441]: Nice catch.

**Review by jeff-cohere — CHANGES_REQUESTED** [review]
Looks good. I have some minor suggest revisions.

File: src/physics_swe.c [hunk 5]
```diff
@@ -267,15 +267,16 @@ static PetscErrorCode RHSFunctionForInternalEdges(RDy rdy, Vec F, AMaxDiagnostic
...
-      }
-
       if (!(hr < tiny_h && hl < tiny_h)) {
         PetscReal areal = cells->areas[l];
         PetscReal arear = cells->areas[r];
 
+        PetscReal cnum = amax_vec_int[ii] * edge_len / fmin(areal, arear) * rdy->dt;
+        if (cnum > courant_num_diags->max_courant_num) {
+          courant_num_diags->max_courant_num = cnum;
+          courant_num_diags->global_edge_id  = edges->global_ids[ii];
```
Comment (donghuix) [inline src/physics_swe.c:277]: How about save the cell id as well? Given the edge id, the max Courant number may occur in either the left cell or the right cell.

File: src/physics_swe.c [hunk 6]
```diff
@@ -505,21 +507,21 @@ PetscErrorCode RHSFunctionSWE(TS ts, PetscReal t, Vec X, Vec F, void *ctx) {
...
-  // write out debugging info for maximum wave speed
+  // write out debugging info for maximum courant number
   if (rdy->config.log_level >= LOG_DEBUG) {
-    MPI_Allreduce(MPI_IN_PLACE, &amax_diags, 1, amax_diags_type, amax_diags_op, rdy->comm);
+    MPI_Allreduce(MPI_IN_PLACE, &courant_num_diags, 1, courant_num_diags_type, courant_num_diags_op, rdy->comm);
     PetscReal dt;
     PetscCall(TSGetTimeStep(ts, &dt));
-    RDyLogDebug(rdy, "Max wave speed %g encountered at edge %d (Courant number = %f)", amax_diags.max_wave_speed, amax_diags.global_edge_id,
-                amax_diags.max_wave_speed * dt * 2);
+    RDyLogDebug(rdy, "Max courant number %g encountered at edge %d (Courant number = %f)", courant_num_diags.max_courant_num,
```
Comment (jeff-cohere) [inline src/physics_swe.c:523]: You can remove the comment in parenthesis here.

**Review by jeff-cohere — APPROVED** [review]

**Review by donghuix — APPROVED** [review]
A minor comment for you to consider.

**Pull request merged**